central/complianceoperator/v2/report/manager/manager_impl.go

```diff
@@ -282,22 +350,56 @@ func (m *managerImpl) handleReadyScan() {
 }
 
 // getScanConfigWatcher returns the ScanConfigWatcher of a given scan
-func (m *managerImpl) getScanConfigWatcher(ctx context.Context, scan *storage.ComplianceOperatorScanV2, ds scanConfigurationDS.DataStore, queue *queue.Queue[*watcher.ScanConfigWatcherResults]) (watcher.ScanConfigWatcher, error) {
-	sc, err := watcher.GetScanConfigFromScan(ctx, scan, ds)
+func (m *managerImpl) getScanConfigWatcher(ctx context.Context, results *watcher.ScanWatcherResults, ds scanConfigurationDS.DataStore, queue *queue.Queue[*watcher.ScanConfigWatcherResults]) (watcher.ScanConfigWatcher, error) {
+	sc, err := watcher.GetScanConfigFromScan(ctx, results.Scan, ds)
```

Comment:
By looking at the name of this function and the comment, I see that the first few lines would do the job and the rest seem to be doing something extra. Shall we rename or split this func?

Reply:
By looking at `log.Debugf("Staring config watcher %s", sc.GetId())`, we could split this into `getConfigWatcher` and `startConfigWatcher`.